File: medium/848.js
var shiftingLetters = function(S, shifts) {
    let sum = 0;
    let newstring = [];
    let alphabet = 'abcdefghijklmnopqrstuvwxyz';
    for (let i = shifts.length-1; i >= 0; i--) {
        sum += shifts[i];
        newstring.unshift(alphabet[(alphabet.indexOf(S[i]) + sum) % 26]);

    }
    return newstring.join("");
};

console.log(shiftingLetters('ruu', [26,9,17]));

File: medium/848.java
class Solution {
    public String shiftingLetters(String S, int[] shifts) {
        String alphabet = "abcdefghijklmnopqrstuvwxyz";
        char[] newstring = new char[S.length()];
        int sum = 0;
        char x;
        for (int i = shifts.length-1; i >= 0; i--) {
            sum += shifts[i] % 26;
            x = S.charAt(i);
            newstring[i] = alphabet.charAt((alphabet.indexOf(x) + sum) % 26);
        }
        return String.valueOf(newstring);
    }
}